hpx/parallel/executors/executor_parameters.hpp

```diff
@@ -0,0 +1,262 @@
+//  Copyright (c) 2016 Marcin Copik
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+/// \file parallel/executors/executor_parameters.hpp
+
+#if !defined(HPX_PARALLEL_EXECUTOR_PARAMETERS)
+#define HPX_PARALLEL_EXECUTOR_PARAMETERS
+
+#include <hpx/config.hpp>
+#include <hpx/traits/is_executor_parameters.hpp>
+#include <hpx/parallel/config/inline_namespace.hpp>
+#include <hpx/parallel/executors/executor_traits.hpp>
+#include <hpx/util/always_void.hpp>
+
+#include <type_traits>
+#include <utility>
+#include <vector>
+#include <cstdarg>
+#include <functional>
+
+#include <boost/ref.hpp>
+namespace hpx { namespace parallel { HPX_INLINE_NAMESPACE(v3)
+{
+    namespace detail {
+
+        template<template<typename> class Condition, typename Arg,
+            typename Enable = void>
+        struct counter_increment
+        {
+            constexpr static int value = 0;
```

Comment:
Please use `HPX_CONSTEXPR` instead of `constexpr`. SOme compilers we support don't handle `constexpr` correctly yet.